languages.toml

```diff
@@ -1835,3 +1836,23 @@ roots = []
 [[grammar]]
 name = "wast"
 source = { git = "https://github.com/wasm-lsp/tree-sitter-wasm", rev = "2ca28a9f9d709847bf7a3de0942a84e912f59088", subpath = "wast" }
+
+
+[[language]]
+name = "groovy"
+scope = "source.groovy"
+injection-regex = "groovy"
+file-types = ["groovy", "gvy", "gy", "gsh"]
+comment-token = "//"
+indent = { tab-width = 2, unit = "  " }
+language-server = { command = "java", args = ["-jar", "/Users/zephaniahong/Desktop/groovy-language-server/build/libs/groovy-language-server-all.jar"]}
+roots = []
+
+[[grammar]]
+name = "groovy"
+source = { git = "https://github.com/tree-sitter/tree-sitter-java", rev = "bd6186c24d5eb13b4623efac9d944dcc095c0dad" }
+
```

Comment:
You either need to use `grammar = "java"` in the `[[language]]` block or use a `tree-sitter-groovy` parser - this will fail to load because we're expecting the parsing to expose symbols under the name `tree_sitter_groovy` but this parser uses `tree_sitter_java`: https://github.com/helix-editor/helix/blob/2457111bf074d4b5c1d45f76e2c7d593ceec5895/helix-loader/src/grammar.rs#L76

Reply:
We could probably raise this log level so it shows up in the log by default, as long as this doesn't start showing up in test output: https://github.com/helix-editor/helix/blob/2457111bf074d4b5c1d45f76e2c7d593ceec5895/helix-core/src/syntax.rs#L371

Reply:
I see! I think I understand now(: Thanks!
Also, what do you mean by 'as long as this doesn't start showing up in test output'? Would changing it to `log::warning!("{}", e)` solve what you're saying?

Reply:
Looking at the test suite, it isn't actually a problem. The integration tests will print log lines but there isn't an integration test where this returns Error

https://github.com/helix-editor/helix/pull/4315